### ISSUE TITLE
ACCEPT_EMPTY_STRING_AS_NULL_OBJECT doesn't work.

### ISSUE THREAD

**Issue by firatkucuk** [FasterXML/jackson-databind#1563]
I tested with 2.8.4 and 2.9.0.pr1 versions.

The expected result is conversion from empty string to null value.

The following code snippet is test case:

```java
public class NullCheckTest {

    @Test
    public void testObjectMapper() throws Exception {

        final HelloForm form = new HelloForm();
        form.setName("");

        final ObjectMapper mapper = new ObjectMapper();
        mapper.enable(DeserializationFeature.ACCEPT_EMPTY_STRING_AS_NULL_OBJECT);

        final String    text    = new ObjectMapper().writeValueAsString(form);
        final HelloForm newForm = mapper.readValue(text, HelloForm.class);

        assert newForm.getName() == null;
    }

    private static class HelloForm {

        String name;

        public String getName() {

            return name;
        }

        public void setName(final String name) {

            this.name = name;
        }
    }
}
```

I tried to use this property in my spring boot project as documented in reference. After a detailed tracking I noticed it's an issue about Jackson.

```properties
spring.jackson.deserialization.accept-empty-string-as-null-object = true
```

**Comment by cowtowncoder** [comment]
First of all, thank you for reporting this: I can understand why you think this is a bug.

However: this will not convert `String` bound values into `null`s; that is not intent. So in this particular case it works as intended.
I will try to improve description of the feature to outline limits of its applicability.

In practical terms the idea is to support coercion into `null` to help map data in cases where it otherwise would not map: and since `String` values are legal for `java.lang.String`, no coercion is intended. Another part of explanation would be that originally intent was really to allow use of "" to construct "empty" POJO instances; and `OBJECT` is to refer to POJOs, although I can see why it could also suggest idea of any `java.lang.Object`.

Having said all that, I would be open to having another `DeserializationFeature` that would specifically allow coercing empty String (but, I presume, not a String that is all whitespace) into `null`.

The reason why I do not think existing behavior should change is that I think that could cause severe backwards-compatibility issues.

**Comment by firatkucuk** [comment]
Thanks for the explanation.

I tried to explain my main problem [so](http://stackoverflow.com/questions/42865437/spring-boot-string-deserializer-for-every-form-field)

An option for converting empty string values to nulls will be good.

**Comment by cowtowncoder** [comment]
For now I think either requiring setter, or custom deserializer, that does that will probably be the best option. Since behavior itself for `ACCEPT_EMPTY_STRING_AS_NULL_OBJECT` is intended, will close this issue; a new one specifically requesting new `DeserializationFeature` may be filed.

**Comment by renannprado** [comment]
I was about to open the same issue. It would be nice to have a `DeserializationFeature` that would automatically convert empty strings to null.

**Comment by renannprado** [comment]
For spring users, I've got an alternative. You can do something like:
```java
@Component
public class StringSanitizerModule extends SimpleModule {
    public StringSanitizerModule() {
        addDeserializer(String.class, new StdScalarDeserializer<String>(String.class) {
            @Override
            public String deserialize(JsonParser jsonParser, DeserializationContext ctx) throws IOException {
                return MyStringUtils.nullIfEmpty(jsonParser.getValueAsString().trim());
            }
        });
    }
}
```
Idea taken from: http://stackoverflow.com/questions/6852213/can-jackson-be-configured-to-trim-leading-trailing-whitespace-from-all-string-pr

**Comment by danielwegener** [comment]
@renannprado nice acceptable solution. But how is this spring related?

**Comment by renannprado** [comment]
It is not. I just left a snippet for people who want to use this solution with spring, hence the `For spring users`. Is that ok?

**Comment by danielwegener** [comment]
@renannprado Makes sense, thanks for the clarification :)

**Comment by VOsipenkov** [comment]
@cowtowncoder did you fix this issue after  2.9.0 release? I faced this issue on 2.11.4 version(integrated in spring-boot autoconfure 2.4.2)

**Comment by cowtowncoder** [comment]
@VOsipenkov no changes were made based on issue reported here. If you have specific problem to report (with a reproduction) against a recent version (2.12.1 ideally, but 2.11.4 could be considered), please file a new issue.
